priv/templates/script.upload.eex

```diff
@@ -125,4 +149,10 @@ if [ "$(uname -s)" = "Darwin" ]; then
     fi
 fi
 
-cat $FILENAME | ssh -s $SSH_OPTIONS $DESTINATION nerves_firmware_ssh2
+# Check if the system is still running the old NervesFirmwareSSH which
+# is listening on port 8989
+if (echo >/dev/tcp/$DESTINATION/8989) &>/dev/null; then
```

Comment:
Wow, I never knew you could do that....  I assume that's a Mac thing?

Anyway, my opinion is that it tries the new way first since that's going to be the right way most of the time. The only time it won't work is when transitioning a device that firmware updates the old way on 8989. I'm not sure if anything can be derived from the error code when trying to update on port 22 and the nerves_firmware_ssh system doesn't exist. If that doesn't provide a hind, maybe a message and a commandline option to update the old way would be sufficient.

Reply:
Sounds good to me. I think this will require `-o ConnectTimeout` ssh option? When I tested trying to let things fail it would just try forever so I think some sort of ssh option will be required to make things stop.

What do you think about taking over the Subsystem name `nerves_firmware_ssh`? I guess if we scope it to `nerves_firmware_ssh2` or something different, we should immediately get back that the Subsystem doesn’t exist

Reply:
K, my bash-fu is too weak right now and I'm tapping out for a bit. I'm trying to be able to check the output message from the first attempt on this subsystem which gives the beautiful message of `subsystem request failed on channel 0`. That way I can only try port 8989 on missing subsystem failure and not _everything_.

But, I can't seem to get stderr where I need it (╯°□°）╯︵ ┻━┻

The ssh failure kills the whole process, so need to trap or wrap in `if` statement, which I can do, but Im failing at programmatically getting the result to be able to properly act on it. Here's what I'm trying. If anything pops out, let me know. Taking a break for now

```shell
if ! cat $FILENAME | ssh -s $SSH_OPTIONS $DESTINATION nerves_firmware_ssh 2>err;
then
    echo $err
fi
```

Reply:
K, my bash-fu is too weak right now and I'm tapping out for a bit. I'm trying to be able to check the output message from the first attempt on this subsystem which gives the beautiful message of `subsystem request failed on channel 0`. That way I can only try port 8989 on missing subsystem failure and not _everything_.

But, I can't seem to get stderr where I need it (╯°□°）╯︵ ┻━┻

The ssh failure kills the whole process, so need to trap or wrap in `if` statement, which I can do, but Im failing at programmatically getting the result to be able to properly act on it. Here's what I'm trying. If anything pops out, let me know. Taking a break for now

```shell
if ! cat $FILENAME | ssh -s $SSH_OPTIONS $DESTINATION nerves_firmware_ssh 2>err;
then
    echo $err
fi
```

Reply:
ugh...ignore me. Figured it out 🤦